whylog/config/rule.py

```diff
@@ -46,7 +72,24 @@ def create_from_intent(cls, user_rule_intent):
             parsers_dict, user_rule_intent
         )
         constraints = cls._create_constraints_list(parser_ids_mapper, user_rule_intent)
-        return Rule(causes, effect, constraints)
+        ordered_causes, modified_constraints = cls._order_causes_list(causes, constraints)
+        return Rule(ordered_causes, effect, modified_constraints)
+
+    @classmethod
+    def _order_causes_list(cls, causes, constraints):
+        causes_with_indexes = [(causes[i], i + 1) for i in six.moves.range(len(causes))]
+        causes_with_indexes.sort(key=lambda x: x[0].name)
+        ordered_causes = []
+        parser_index_mapping = {}
+        for i in six.moves.range(len(causes_with_indexes)):
```

Comment:
Can not, because original numeration has been changed by sorting. But I simplified this code a bit.